packages/api/amplify_api_dart/lib/src/decorators/web_socket_auth_utils.dart

```diff
@@ -36,14 +41,38 @@ Future<Uri> generateConnectionUri(
   );
   final encodedAuthHeaders =
       base64.encode(json.encode(authorizationHeaders).codeUnits);
-  final endpointUri = Uri.parse(
-    config.endpoint.replaceFirst('appsync-api', 'appsync-realtime-api'),
-  );
-  return Uri(scheme: 'wss', host: endpointUri.host, path: 'graphql').replace(
-    queryParameters: <String, String>{
-      'header': encodedAuthHeaders,
-      'payload': base64.encode(utf8.encode(json.encode(_emptyBody))),
-    },
+  final authQueryParameters = {
+    'header': encodedAuthHeaders,
+    'payload': base64.encode(utf8.encode(json.encode(_emptyBody))),
+  };
+  // Conditionally format the URI for a) AppSync domain b) custom domain. In either
+  // case, add the auth query parameters.
+  final endpointUriHost = Uri.parse(config.endpoint).host;
+  if (endpointUriHost.contains(_appSyncHostPortion) &&
+      endpointUriHost.endsWith('amazonaws.com')) {
+    // AppSync domain that contains "appsync-api" and ends with "amazonaws.com."
+    // Replace "appsync-api" with "appsync-realtime-api," append "/graphql" and
+    // add formatted auth query parameters.
+    final appSyncRealtimeUri = Uri.parse(
+      config.endpoint
+          .replaceFirst(_appSyncHostPortion, _appSyncRealtimeHostPortion),
+    );
+    return Uri(
+      scheme: _wssScheme,
+      host: appSyncRealtimeUri.host,
+      path: 'graphql',
+    ).replace(
+      queryParameters: authQueryParameters,
+    );
+  }
+  // Custom domain (not AppSync); format URL as documented on https://docs.aws.amazon.com/appsync/latest/devguide/custom-domain-name.html.
+  // Appends "/graphql/realtime" path in addition to auth query parameters.
+  return Uri(
+    scheme: _wssScheme,
+    host: endpointUriHost,
+    path: 'graphql/realtime',
```

Comment:
This only differs from the above construction by `path` - can you consolidate the return so that's clear and there's no drift?

Reply:
tried to make that a bit clearer and only use the `Uri` constructor once with modified variables from if/else for appsync/custom domain.